### PR TITLE
Zklnd 59 - Use Go API for changing channel status

### PR DESCRIPTION
**Pull request by DariusParvin**
#### Pull Request Checklist

- [ ] If this is your first time contributing, we recommend you read the [Code
  Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
- [ ] All changes are Go version 1.12 compliant
- [ ] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
- [ ] For new code: Code is accompanied by tests which exercise both
  the positive and negative (error paths) conditions (if applicable)
- [ ] For bug fixes: Code is accompanied by new tests which trigger
  the bug being fixed to prevent regressions
- [ ] Any new logging statements use an appropriate subsystem and
  logging level
- [ ] Code has been formatted with `go fmt`
- [ ] Protobuf files (`lnrpc/**/*.proto`) have been formatted with
  `make rpc-format` and compiled with `make rpc`
- [ ] For code and documentation: lines are wrapped at 80 characters
  (the tab character should be counted as 8 characters, not 4, as some IDEs do
  per default)
- [ ] Running `make check` does not fail any tests
- [ ] Running `go vet` does not report any issues
- [ ] Running `make lint` does not report any **new** issues that did not
  already exist
- [ ] All commits build properly and pass tests. Only in exceptional
  cases it can be justifiable to violate this condition. In that case, the
  reason should be stated in the commit message.
- [ ] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)

### CODE REVIEW

File: zkbreacharbiter_test.go [hunk 1]
```diff
@@ -171,6 +171,45 @@ func initTestMerchState(DBPath string, skM string, payoutSkM string, disputeSkM
...
+	if err != nil {
+		zkchLog.Error(err)
+		return err
+	}
+
+	// Flip escrowTxid to Big Endian to match how it is stored in
+	// merchState.ChannelStatusMap
+	//This works because hex strings are of even size
+	s := ""
+	for i := 0; i < len(escrowTxid)/2; i++ {
```
Comment (gijsvl) [inline zkbreacharbiter_test.go:191]:
```
for i := 0; i < len(escrowTxid); i += 2 {
    s = escrowTxid[i:i+2] + s
}
```

File: zkbreacharbiter_test.go [hunk 2]
```diff
@@ -263,8 +306,8 @@ func testZkBreachSpends(t *testing.T, test breachTest) {
 		t.Fatalf("breach arbiter didn't send ack back")
 	}
 
-	notifier := brar.cfg.Notifier.(*mockSpendNotifier)
-	notifier.confChannel <- &chainntnfs.TxConfirmation{}
+	// notifier := brar.cfg.Notifier.(*mockSpendNotifier)
```
Comment (gijsvl) [inline zkbreacharbiter_test.go:309]: Is there a reason to leave this as a comment? Are we gonna uncomment it at some point?

File: zkbreacharbiter_test.go [hunk 3]
```diff
@@ -297,9 +340,8 @@ func testZkBreachSpends(t *testing.T, test breachTest) {
 		t.Fatalf("disputeTx fee of %v sat is less than minRelayTxFee of %v sat", disputeTxFee, minTxFee)
 	}
 
-	// Deliver confirmation of sweep if the test expects it.
-	if test.sendFinalConf {
-		notifier.confChannel <- &chainntnfs.TxConfirmation{}
-	}
-
+	// // Deliver confirmation of sweep if the test expects it.
+	// if test.sendFinalConf {
```
Comment (gijsvl) [inline zkbreacharbiter_test.go:344]: Same... Better remove if we don't need it anymore.

File: zkchannelmanager.go [hunk 4]
```diff
@@ -2448,23 +2513,177 @@ func (z *zkChannelManager) CloseZkChannel(notifier chainntnfs.ChainNotifier, zkC
...
+	merchState, err := zkchanneldb.GetMerchState(zkMerchDB)
+	if err != nil {
+		zkchLog.Error(err)
+		return "", err
+	}
+
+	// Flip bytes from Little Endiand to Big Endian
+	// This works because hex strings are of even size
+	s := ""
+	for i := 0; i < len(escrowTxid)/2; i++ {
```
Comment (gijsvl) [inline zkchannelmanager.go:2675]: Same as previous comment, this can be made more readable

File: zkbreacharbiter_test.go (hunk 3 above)
Comment (DariusParvin) [reply zkbreacharbiter_test.go:344]: has been uncommented

File: zkbreacharbiter_test.go (hunk 2 above)
Comment (DariusParvin) [reply zkbreacharbiter_test.go:309]: removed

File: zkbreacharbiter_test.go (hunk 1 above)
Comment (DariusParvin) [reply zkbreacharbiter_test.go:191]: nice. All cases of this have changed now

**Pull request merged**